docs/sql/statements/insert.rst

```diff
@@ -84,9 +84,12 @@ the explicit or implicit column list left-to-right.
 CrateDB will not fill any column not present in the explicit or implicit column
 list.
 
-If the :ref:`expression <gloss-expression>` for any column is not of the
-correct data type, CrateDB will attempt automatic :ref:`type conversion
-<data-types-casting>`.
+.. CAUTION::
+
+    If the values for any column are not of the correct data type, CrateDB
+    will attempt automatic :ref:`type conversion <data-types-casting>`,
+    but if the casting fails the row will be skipped and the ``INSERT``
```

Comment:
Did a couple more tests

Using VALUES type casting still happens automatically, but we do get a SQLParseException if any of the values fails casting, and nothing gets inserted --> I need to adjust the wording

Using SELECT the behaviour seems consistent except for int to bit, see below:
- text (not numeric) to int --> silent failure
- text longer than 32766 chars to text with column store --> silent failure
- text (invalid date) to timestamp --> silent failure
- unnest(['not a number', '99']) to int --> silent failure, 99 gets inserted
- text (valid) to geo_shape / geo_point --> silent failure 
- text (valid) to array(text) --> silent failure 
- text to object --> silent failure
- text (invalid) to IP --> silent failure
- int (0) to bit(1) --> SQLParseException 
- text to bit(1) --> silent failure
- text to boolean --> silent failure
- long to int --> silent failure
- double to int --> silent loss of precision
- text (double) to int --> silent failure

Reply:
Afaik it also fails silently with column constraints and generated columns.
I think it makes sense, that the db team review this.

Reply:
The general behaviour is related to non-existing transactions at CrateDB. Such if one writes multiple rows, and the error happened e.g. at the 2nd row, the 1st row is already written and cannot be rolled back. Thus we just decrease the affected row count at the response instead of raising an exception. We document this general behaviour here https://crate.io/docs/crate/reference/en/5.0/general/dml.html#inserting-data. I'd suggest to either link to this or phrase it in a similar way as this is not only related to type casting, but also e.g. type validation when inserting a string exceeding the allowed chars of a varchar(N) type.

Reply:
The `int (0) to bit(1) --> SQLParseException` looks also valid as we do not support implicit type casting from `int` to `bit`.

Reply:
That is correct, yes, but the dml page currently reads "When inserting a single row, if an error occurs an error is returned as a response", but in reality it fails silently also for single rows, do we want to consider that a bug or should I review the wording on that page as well?

Reply:
I think it fails silently for `INSERT ... SELECT` even if the select returns a single row, right? But that's related to that if the select will return a single row or more is not known.

Reply:
The dml page also talks about valid rows still getting inserted right after an example with VALUES where in reality all rows are validated before any value is inserted

Reply:
Makes sense, thank you @seut , I will review both pages and re-request review a bit later today.